bundles/org.openhab.binding.mielecloud/src/test/java/org/openhab/binding/mielecloud/internal/util/EmailValidatorTest.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.mielecloud.internal.util;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+/**
+ * @author Björn Lange - Initial Contribution
+ */
+@NonNullByDefault
+public class EmailValidatorTest {
+    @ParameterizedTest
+    @ValueSource(strings = { "example@openhab.org", "itsme@test24.com", "my-account@t-online.de" })
```

Comment:
Maybe add testcases for the actual changes in this PR, for example allowing capital letters?

Reply:
I added some further test cases regarding the minimum length and uppercase letters. However, the previously already present test cases checked the style of the addresses that were reported to be problematic, namely ones that have numbers and dashes. Also capital letters are part of Java's `\w` shortcut `Pattern` syntax, so nothing should have changed in that regard in the backend validation. The frontend validation differed from that and I decided that it would be easier to understand if the regex in front- and backend would be as similar as possible and thus replaced `\w`.